lib-ee/emqx_ee_bridge/test/emqx_ee_bridge_gcp_pubsub_SUITE.erl

```diff
@@ -0,0 +1,1454 @@
+    assert_metrics(
+        #{
+            batching => 0,
+            dropped => 0,
+            failed => 1,
+            inflight => 0,
+            matched => 1,
+            queuing => 0,
+            retried => 0,
+            success => 0
+        },
+        ResourceId
+    ),
+    ok.
+
+t_failed_to_start_jwt_worker(Config) ->
+    ?check_trace(
+        emqx_common_test_helpers:with_mock(
+            emqx_connector_jwt_sup,
+            ensure_worker_present,
+            fun(_JWTWorkerId, _Config) -> {error, restarting} end,
+            fun() ->
+                ?assertMatch({ok, _}, create_bridge(Config))
+            end
+        ),
+        fun(Trace) ->
+            ?assertMatch(
+                [#{reason := {error, restarting}}],
+                ?of_kind(gcp_pubsub_bridge_jwt_worker_failed_to_start, Trace)
+            ),
+            ok
+        end
+    ),
+    ok.
+
+t_stop(Config) ->
+    Name = ?config(gcp_pubsub_name, Config),
+    {ok, _} = create_bridge(Config),
+    ?check_trace(
+        ?wait_async_action(
+            emqx_bridge_resource:stop(?BRIDGE_TYPE, Name),
+            #{?snk_kind := gcp_pubsub_stop},
+            5_000
+        ),
+        fun(Res, Trace) ->
+            ?assertMatch({ok, {ok, _}}, Res),
+            ?assertMatch([_], ?of_kind(gcp_pubsub_stop, Trace)),
+            ok
+        end
+    ),
+    ok.
+
+t_get_status_ok(Config) ->
+    ResourceId = ?config(resource_id, Config),
+    {ok, _} = create_bridge(Config),
+    ?assertEqual({ok, connected}, emqx_resource_manager:health_check(ResourceId)),
+    ok.
+
+t_get_status_no_worker(Config) ->
+    ResourceId = ?config(resource_id, Config),
+    {ok, _} = create_bridge(Config),
+    emqx_common_test_helpers:with_mock(
+        ehttpc,
+        workers,
+        fun(_Poolname) -> [] end,
+        fun() ->
+            ?assertEqual({ok, disconnected}, emqx_resource_manager:health_check(ResourceId)),
+            ok
+        end
+    ),
+    ok.
+
+t_get_status_down(Config) ->
+    ResourceId = ?config(resource_id, Config),
+    {ok, _} = create_bridge(Config),
+    emqx_common_test_helpers:with_mock(
+        ehttpc,
+        health_check,
+        fun(_Worker, _Timeout) ->
+            {error, connect_timeout}
+        end,
+        fun() ->
+            ?assertEqual({ok, disconnected}, emqx_resource_manager:health_check(ResourceId)),
+            ok
+        end
+    ),
+    ok.
+
+t_get_status_timeout_calling_workers(Config) ->
+    ResourceId = ?config(resource_id, Config),
+    {ok, _} = create_bridge(Config),
+    emqx_common_test_helpers:with_mock(
+        ehttpc,
+        health_check,
+        fun(_Worker, _Timeout) ->
+            receive
+            after infinity -> error(impossible)
+            end
+        end,
+        fun() ->
+            ?assertEqual({ok, disconnected}, emqx_resource_manager:health_check(ResourceId)),
+            ok
+        end
+    ),
+    ok.
+
+t_on_start_ehttpc_pool_already_started(Config) ->
```

Comment:
NIT: I am not sure, here we have some quite complicated cases to test single lines in GCP connector... Do we really need to do this through the whole bridge stack?

Reply:
I wanted to have the most coverage I could get, and preferably in an "integration test" way, so I could observe the overall behavior of the bridge under failures.